Tasks/AzureMonitorAlerts/AuthorizationClient.ts

```diff
@@ -0,0 +1,86 @@
+import * as tl from "vsts-task-lib/task";
+import * as Q from "q";
+import * as querystring from "querystring";
+import { 
+	HttpClient, 
+	HttpClientResponse 
+} from "typed-rest-client/HttpClient";
+
+export interface IAccessToken {
+	token_type: string;
+	expires_in: string;
+	expires_on: string;
+	resource: string;
+	access_token: string;
+
+	// ignored other properties as they won't be useful in the current scenario
+}
+
+export class AuthorizationClient {
```

Comment:
How different is this from azure-arm-common?

Reply:
In azure-arm-common, we fetched access token for every rest call that we made. Since the access token has a validity of 3600 sec, fetching access token per network call should be avoided. 

The current AuthorizationClient fetches the token once and stores it. A fresh access token is fetched only when the access token is close to expiry (60 sec window).